src/AnalysisPrograms/Recognizers/PteropusSpecies.cs

```diff
@@ -0,0 +1,450 @@
+        /// <param name="genericConfig">config file that contains parameters used by all profiles.</param>
+        /// <param name="segmentStartOffset">when recording starts.</param>
+        /// <param name="getSpectralIndexes">not sure what this is.</param>
+        /// <param name="outputDirectory">where the recogniser results can be found.</param>
+        /// <param name="imageWidth"> assuming ????.</param>
+        /// <returns>recogniser results.</returns>
+        public override RecognizerResults Recognize(AudioRecording audioRecording, Config genericConfig, TimeSpan segmentStartOffset, Lazy<IndexCalculateResult[]> getSpectralIndexes, DirectoryInfo outputDirectory, int? imageWidth)
+        {
+            // Examples of the APIs available. You don't need all of these commands! Pick and choose.
+            if (ConfigFile.HasProfiles(genericConfig))
+            {
+                string[] profileNames = ConfigFile.GetProfileNames<Config>(genericConfig);
+                int count = profileNames.Length;
+                var message = new StringBuilder("Found " + count + " config profile(s): ");
+                foreach (string s in profileNames)
+                {
+                    message.Append(s + ", ");
+                }
+
+                log.Info(message.ToString());
+            }
+            else
+            {
+                log.Warn("No configuration profiles found. Three profiles expected for the Flying Fox recogniser.");
+            }
+
+            RecognizerResults results1 = null;
+
+            if (ConfigFile.TryGetProfile(genericConfig, "Territorial", out var profile1))
+            {
+                results1 = TerritorialCall(audioRecording, genericConfig, profile1, outputDirectory, segmentStartOffset);
+                log.Info("Territory event count = " + results1.Events.Count);
+            }
+            else
+            {
+                log.Warn("Could not access Territorial configuration parameters");
+            }
+
+            RecognizerResults results2 = null;
+            if (ConfigFile.TryGetProfile(genericConfig, "Wingbeats", out var profile2))
+            {
+                results2 = WingBeats(audioRecording, genericConfig, profile2, outputDirectory, segmentStartOffset);
+                log.Info("Wingbeat event count = " + results2.Events.Count);
+            }
+            else
+            {
+                log.Warn("Could not access Wingbeats configuration parameters");
+            }
+
+            // combine the results
+            if (results1 != null && results2 != null)
+            {
+                results1.Events.AddRange(results2.Events);
+                results1.Plots.AddRange(results2.Plots);
+            }
+
+            //Set following true if you want special debug spectrogram, i.e. with special plots
+            //In addition, standard spectrograms are produced when you set true in the config file, Towsey.PteropusSpecies.yml.
+            if (false)
```

Comment:
this practice is messy. Didn't we augment `RecognizerResults` so that the normal debug spectrogram can draw additional tracks?

Remove if not needed, otherwise, cleaning this up is a job for another time